### DISCUSSION TITLE
implementation of `Spinner()` and `ProgressIndicator()`

### DISCUSSION THREAD

**Discussion by kyle-power** [Appsilon/shiny.fluent#38] [Q&A] [answered]
@kamilzyla 
What is the suggested method of using `Spinner()` and `ProgressIndicator()` as dynamic indicators? For example, when a `PrimaryButton` is clicked, render a loading `Spinner` until the server function completes, then hide spinner?

I've tried with `renderUI()`, however as `{R}` is single-threaded, enabling and disabling a `uiOutput` in the same `observeEvent()` function doesn't render anything on the page.

Any suggestions?

**Comment by marekrogala** [comment]
@kyle-power thanks, I think such an example would be super valuable. We've just made significant improvements to the package, so this should be easier in the new version. We'll try prepare an example for you soon.

**Comment by anirbanshaw24** [comment]
@kyle-power You are looking for something like [this](https://shiny.rstudio.com/gallery/progress-bar-example.html) right?
The shiny.fluent package is pretty new and we are working on making this robust and simple. We will come back with a solution soon, mostly within a month.
Stay tuned! :)

**Reply by kyle-power** [reply]
@anirbanshaw24 Yes, exactly. I think that is a full page loader, so I think it would be beneficial to also have a small spinner, like here: https://developer.microsoft.com/en-us/fluentui#/controls/web/spinner
As a visual indicator that the server is working on something. Thanks so much!!

**Comment by kamilzyla** [comment, marked as answer]
@kyle-power After investigating the problem a bit, I have to admit that it's quite difficult to use custom progress indicators. I created an issue to find a solution for that here: appsilon/shiny.react#22. You can also check it out for a partial workaround (for a simple busy / not busy  case).
